### PR TITLE
Improve assertion docs

### PR DESCRIPTION
**Pull request by keynmol**
1) TL;DR of expectations on top
2) Detailed examples showing the output/purpose/best practices of various expectations
3) Adding suite outputs to some other pages

See the full result here: http://weaver-site.surge.sh/weaver-test/docs/expectations

### CODE REVIEW

**Comment by Baccata** [comment]
That looks really neat !

**Review by Baccata — APPROVED** [review]

File: docs/resources.md [hunk 1]
```diff
@@ -45,3 +45,6 @@ object HttpSuite extends IOSuite {
 }
 ```
 
+```scala mdoc:passthrough
+println(weaver.docs.Output.runSuites(HttpSuite).unsafeRunSync())
```
Comment (keynmol) [inline docs/resources.md:49]: One worry is that this test suite actually runs an HTTP request to httpbin...

I'll probably revert this, don't want it to break during CI.

File: modules/docs/src/main/scala/weaver/FansiModifier.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package weaver
+package docs
+
+import mdoc._
+
+class FansiModifier extends mdoc.PostModifier {
```
Comment (keynmol) [inline modules/docs/src/main/scala/weaver/FansiModifier.scala:6]: This is not actually used, but I struggle to part ways with it, given how ugly of a hack it is.

File: website/static/css/custom.css [hunk 3]
```diff
@@ -49,4 +49,5 @@ div.terminal pre {
   background-color: black;
   color: white;
   padding: 20px;
+  overflow: auto
```
Comment (keynmol) [inline website/static/css/custom.css:52]: That's for the ScalaCheck integration page: http://weaver-site.surge.sh/weaver-test/docs/scalacheck 

The error message is hella long and breaks rendering.

File: docs/resources.md (hunk 1 above)
Comment (keynmol) [reply docs/resources.md]: Changed in 3fd0aee, arguably better example as a whole: http://weaver-site.surge.sh/weaver-test/docs/resources

**Pull request merged**